apps/ios/SimpleX NSE/NotificationService.swift

```diff
@@ -137,24 +223,48 @@ class NSEThreads {
         }
     }
 
+    @inline(__always)
     var noThreads: Bool {
         allThreads.isEmpty
     }
 }
 
-struct ExpectedMessage {
+// ExpectedMessage is a bad name, because it containes expectedMsg inside.
+// It is the state for a processor for connection entity that expects a message.
+// Single NSE instance may have several of them.
```

Comment:
```suggestion
// NotificationEntity is a processing state for notifications from a single connection entity (message queue).
// Each NSE instance within NSE process can have more than one NotificationEntity.
// NotificationEntities of an NSE instance are processed concurrently, as messages arrive in any order.
// NotificationEntities for the same connection across multiple NSE instances (NSEThreads) are processed sequentially, so that the earliest NSE instance receives the earliest messages.
// The reason for this complexity is to process all required messages within allotted 30 seconds,
// accounting for the possibility that multiple notifications may be delivered concurrently. 
```